ambry-tools/src/main/java/com.github.ambry/tools/admin/ServerAdminTool.java

```diff
@@ -781,16 +787,40 @@ private ByteBuffer sendRequestGetResponse(DataNodeId dataNodeId, Send request) t
       }
       List<ResponseInfo> responseInfos = networkClient.sendAndPoll(requestInfos, POLL_TIMEOUT_MS);
       if (responseInfos.size() > 1) {
+        // May need to relax this check because response list may contain more than 1 response
         throw new IllegalStateException("Received more than one response even though a single request was sent");
       } else if (!responseInfos.isEmpty()) {
         responseInfo = responseInfos.get(0);
       }
-      requestInfos = Collections.EMPTY_LIST;
+      requestInfos = Collections.emptyList();
     } while (responseInfo == null);
     if (responseInfo.getError() != null) {
       throw new IllegalStateException(
           identifier + ": Encountered error while trying to send request - " + responseInfo.getError());
     }
     return responseInfo.getResponse();
   }
+
+  /**
+   * Get replica of given {@link PartitionId} from given {@link DataNodeId}. If partitionId is null, it returns any
+   * replica on the certain node.
+   * @param dataNodeId the {@link DataNodeId} on which replica resides.
+   * @param partitionId the {@link PartitionId} which replica belongs to.
+   * @return {@link ReplicaId} from given node.
+   */
+  private ReplicaId getReplicaFromNode(DataNodeId dataNodeId, PartitionId partitionId) {
+    ReplicaId replicaToReturn = null;
+    if (partitionId != null) {
+      for (ReplicaId replicaId : partitionId.getReplicaIds()) {
+        if (replicaId.getDataNodeId().getHostname().equals(dataNodeId.getHostname())) {
+          replicaToReturn = replicaId;
+          break;
+        }
+      }
+    } else {
+      // pick any replica on this node
+      replicaToReturn = clusterMap.getReplicaIds(dataNodeId).get(0);
```

Comment:
Why pick up any replica works ?

Reply:
Note that in `ServerAdminTool`, there are several methods allowing `PartitionId` to be `null`. In this case, it doesn't matter which replica in RequestInfo because it is never used. The admin request would be issued against all partitions on the node. So, we can pick any replica to instantiate the RequestInfo.